website/docs/subform.md

```diff
@@ -0,0 +1,269 @@
+The typical use case is when you have a field which is a variable list of complex elements.
+
+For example, let's say you have a form asking the user their `name`, `surname` and `familyMembers`. The `familyMembers`, in turn, have their own fields (`name` and `surname`) that need to be validated _before_ being added to the user's field array.
+
+```twoslash include textfield
+import { FieldProps, NonEmptyArray } from "@buildo/formo";
+
+type Props = FieldProps<string, string, NonEmptyArray<unknown>>;
+
+export function TextField(props: Props) {
+  return (
+    <div>
+      <label>{props.label}</label>
+      <input
+        name={props.name}
+        type="text"
+        value={props.value}
+        onChange={(e) => props.onChange(e.currentTarget.value)}
+        onBlur={props.onBlur}
+        disabled={props.disabled}
+      />
+    </div>
+  );
+}
+```
+
+```tsx twoslash
+// @include: textfield
+// ---cut---
+import { useFormo, subFormValue, success, validators } from "@buildo/formo";
+
+type FamilyMember = {
+  name: string;
+  surname: string;
+};
+
+export const MyForm = () => {
+  const { handleSubmit, fieldProps, subForm, formErrors } = useFormo(
+    {
+      initialValues: {
+        name: "Jhon",
+        surname: "Doe",
+        familyMembers: subFormValue([] as Array<FamilyMember>),
+      },
+      fieldValidators: () => ({}),
+      subFormValidators: () => ({
+        familyMembers: {
+          name: validators.fromPredicate(
+            (i) => typeof i === "string" && i.length > 0,
+            "Family member name is required"
+          ),
+          surname: validators.fromPredicate(
+            (i) => typeof i === "string" && i.length > 0,
+            "Family member surname is required"
+          ),
+        },
+      }),
+    },
+    {
+      onSubmit: () => Promise.resolve(success(null)),
+    }
+  );
+
+  const emptyFamilyMember = {
+    name: "",
+    surname: "",
+  };
+
+  return (
+    <div>
+      <TextField label="name" {...fieldProps("name")} />
+      <TextField label="surname" {...fieldProps("surname")} />
+
+      <div>
+        <button
+          onClick={() => subForm("familyMembers").push(emptyFamilyMember)}
+        >
+          Add family members
+        </button>
+
+        {subForm("familyMembers").items.map((familyMember, index) => (
+          <div key={`${familyMember.namePrefix}-container`}>
+            <TextField
+              label={`${index}-name`}
+              {...familyMember.fieldProps("name")}
+            />
+            <TextField
+              label={`${index}-surname`}
+              {...familyMember.fieldProps("surname")}
+            />
+            <p>
+              {familyMember.fieldProps("name").issues}
+              <br />
+              {familyMember.fieldProps("surname").issues}
+            </p>
+          </div>
+        ))}
+      </div>
+
+      <button onClick={handleSubmit}>Sign up</button>
+    </div>
+  );
+};
+```
+
+## Initialization
```

Comment:
I'm not sure I understand the title of this section, maybe "Define a subForm" or "Initializing a subForm"? Also, I would put the phrase "Let's break down the code above." before this section title, and not below it (as this is the first step in breaking it down if I understand correctly)
<img width="802" alt="Screenshot 2022-03-30 at 12 00 03" src="https://user-images.githubusercontent.com/2643520/160805951-a0594665-8a08-43cf-83e8-62083042e4e0.png">